hyperspy/drawing/image.py

```diff
@@ -377,7 +378,11 @@ def plot(self, data_function_kwargs={}, **kwargs):
     def _add_colorbar(self):
         # Bug extend='min' or extend='both' and power law norm
         # Use it when it is fixed in matplotlib
-        self._colorbar = plt.colorbar(self.ax.images[0], ax=self.ax)
+        if self.mesh:
+            ims = self.ax.collections
+        else:
+            ims = self.ax.images
```

Comment:
```suggestion
        ims = self.ax.images if len(self.ax.images) else self.ax.collections:
```

Maybe make it a private property too, since this is being accessed from 2 different places?